### PR TITLE
refactor(dashboard): port to Polymer 3; stop using bower entirely

### PR DESCRIPTION
**Pull request by Lange**
Ready for review. Bower has been entirely removed, and the docs have been updated to reflect this change.

I've also removed the `polymer build` process, which was used to somewhat minify the Dashboard. I removed this process for the following reasons:

1. it barely worked to begin with, and now it seems even harder to get working again after this change
2. now that we're using ES Modules for everything, we can probably just use rollup or something instead of this Polymer-specific tooling
3. `polymer build` has big problems with source maps

If we decide that we want a bundling/minification step again, we can evaluate something like rollup.

### CODE REVIEW

**Comment by Lange** [comment]
All deps have been moved off bower, but we need to QA every single page (including all the weird little error pages) to ensure they still work.

**Comment by Lange** [comment]
Tests are behaving weird. Looks like they can at least sometimes pass on CI, but I have yet to get them to pass locally on my Windows box.

**Comment by Lange** [comment]
One of the flaky tests is:
- `singleInstance - should allow the graphic to be taken after being killed`

Working on trying to figure out why its flaky

**Comment by Lange** [comment]
The problem is really hard to catch. If I run only a subset of tests (`ava test/registration.js`), then the failure almost never happens. It only seems to happen when all tests are running at max parallelism (`ava test`).

**Comment by Lange** [comment]
Furthermore, the failure almost never happens with `headless: false`. This is annoying, because it's much harder to debug with `headless: true`.

**Comment by Lange** [comment]
I'm now getting page crash errors like this randomly:

```
Unhandled Rejection: test\api\client-graphic.js
  Error: Page crashed!
  Page._onTargetCrashed (node_modules/puppeteer/lib/Page.js:215:24)
CDPSession.Page.client.on.event (node_modules/puppeteer/lib/Page.js:123:56)
CDPSession._onMessage (node_modules/puppeteer/lib/Connection.js:200:12)
Connection._onMessage (node_modules/puppeteer/lib/Connection.js:112:17)
WebSocketTransport._ws.addEventListener.event (node_modules/puppeteer/lib/WebSocketTransport.js:44:24)
WebSocket.onMessage (node_modules/ws/lib/event-target.js:120:16)
Receiver.receiverOnMessage (node_modules/ws/lib/websocket.js:789:20)
Receiver.dataMessage (node_modules/ws/lib/receiver.js:422:14)
Receiver.getData (node_modules/ws/lib/receiver.js:352:17)
Receiver.startLoop (node_modules/ws/lib/receiver.js:138:22)
Receiver._write (node_modules/ws/lib/receiver.js:74:10)
Socket.socketOnData (node_modules/ws/lib/websocket.js:864:35)
```

Pretty big bummer, I have no idea what to do about this.

**Comment by Lange** [comment]
_sigh_, setting `--concurrency 1` in ava (`ava --concurrency 1 test`) seems to make these failures go away.

Going to play around with this a bit more before completely resigning myself to slower tests.

**Comment by Lange** [comment]
Well now I'm just mad lmao -- it passes locally but is failing on CI.

**Comment by Lange** [comment]
@Hoishin oh I see, I'll revert that part

**Comment by Lange** [comment]
@Hoishin I think I have this working now -- going to re-run tests on travis a bunch of times to ensure it's not flaky.

My question to you is: do you mind this being one massive PR which kinda does several things? Breaking this up into smaller PRs would be possible, but a huge pain in the butt and maybe not worth the hours it would take me.

**Comment by Lange** [comment]
Okay, I re-ran the tests on Travis like 20 times and they passed every time. Flakiness appears to be resolved.

**Comment by Seldszar** [comment]
Tried with one of my previous bundles: everything is working so far.

👍

**Comment by Dan-Shields** [comment]
tested using my most recent bundle (https://github.com/UoMEsports/HiveAid-bundle) and all seems to look good. Only feature I don't really use in the dashboard is assets, other than that, panels, dialogs, mixer etc. all work same as before 👍

**Comment by Lange** [comment]
@Seldszar @Dan-Shields Thanks for your testing! The Assets system is pretty thoroughly covered by our test suite, so I'm confident that it works as well.

**Comment by Lange** [comment]
Regarding the reduced concurrency: AVA already only runs at a concurrency of 2 in CI environments, due to it apparently not being possible to reliably determine the number of CPU cores available in many CI hosts: https://github.com/avajs/ava/commit/3f81fc431016ca115177d2d79cf9db36b05f69ad

Dropping our concurrency from 2 to 1 here does not seem to have greatly slowed the tests down.

For local dev, this loss of concurrency has more of an impact, but honestly it still doesn't seem huge? For now, I'm enjoying the fact that our tests finally appear to be flake-free. We can introduce concurrency back at a later date.

File: lib/dashboard.js [hunk 1]
```diff
@@ -33,51 +33,47 @@ app.get('/dashboard', ncgUtils.authCheck, (req, res) => {
...
+					return next();
+				}
+			});
+		}
+
+		return next();
+	});
 } else {
-	app.use('/', express.static(BUILD_PATH));
+	log.warn('Serving unbundled files for development');
```
Comment (Hoishin) [inline lib/dashboard.js]: I think this will be logged every time NodeCG is launched without NODECG_TEST environment variable.

Comment (Lange) [reply lib/dashboard.js]: Good catch, I had meant to remove this log line completely. Fixed.

**Comment by Lange** [comment]
@Hoishin Yeah that's a good point. I _think_ I want to take the following route:

- Not consider this a breaking change, because it doesn't break _bundles_
- But, yes we need to also release a new version of `nodecg-cli` to remove this: https://github.com/nodecg/nodecg-cli/blob/master/src/commands/setup.ts#L168
- `generator-nodecg` should be fine as-is, because it doesn't install the top-level NodeCG deps (I don't think it does any dependency installations at all)

If that sounds good to you, I'll go ahead an open a PR on `nodecg-cli` as well, and will merge both PRs at the same time.

**Comment by Hoishin** [comment]
That sounds good, except that nodecg-cli should probably support both cases for the time being (i.e. look for bower.json and run `bower i` if it exists).

**Pull request merged**